test/cpp/interop/interop_client.cc

```diff
@@ -100,27 +104,20 @@ InteropClient::ServiceStub::GetUnimplementedServiceStub() {
   return unimplemented_service_stub_.get();
 }
 
-void InteropClient::ServiceStub::Reset(
-    const std::shared_ptr<Channel>& channel) {
-  channel_ = channel;
-
-  // Update stub_ as well. Note: If new_stub_every_call_ is true, we can reset
-  // the stub_ since the next call to Get() will create a new stub
-  if (new_stub_every_call_) {
-    stub_.reset();
+void InteropClient::ServiceStub::ResetChannel() {
+  channel_ = channel_creation_func_(test_case_);
+  if (!new_stub_every_call_) {
```

Comment:
I don't the update here. Do I miss anything?

Reply:
If `new_stub_every_call_` is true, then there is no need to do anything to the stub, since next time someone calls `Get()` a new one will be created. It is only if `new_stub_every_call_` is false that we explicitly create new stub